### PR TITLE
add home_work_decorator

### PR DESCRIPTION
**Pull request by Kateryna-hub**

### CODE REVIEW

File: HW_decorator.py [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+
+
+def check_type_of_args(func):
+    @wraps(func)
+    def wrapper(*args):
+        result = ''
+        if type(*args) is str:
+            raise ValueError('string type is not supported')
+        if type(*args) is int:
+            result = func(*args)
```
Comment (katevoskresenska) [inline HW_decorator.py:33]: It is better to use isinstance(args, int), isinstance(args, str)

File: HW_decorator.py [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+# ЗАДАЧА-2
+# Написать декоратор который будет выполнять предпроверку типа аргумента который передается в вашу функцию.
+# Если это int, тогда выполнить функцию и вывести результат, если это str(),
+# тогда зарейзить ошибку ValueError (raise ValueError(“string type is not supported”))
+
+
+def check_type_of_args(func):
+    @wraps(func)
+    def wrapper(*args):
+        result = ''
```
Comment (katevoskresenska) [inline HW_decorator.py:29]: You can avoid using this var: result = '' . Just add one more indent to the 'return result' line

File: HW_decorator.py [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+                    print(f'Used cache with counter = {num}')
+
+                else:
+                    cache.update({args: func(*args)})
+                    print(f'Function executed with counter = {num}, function result = {cache[args]}')
+
+            return f'\nThe function was executed {number} times'
+        return wrapper
+    return cache_args_result
+
```
Comment (katevoskresenska) [inline HW_decorator.py:61]: Here is the output I get calling your function:
Function executed with counter = 0, function result = 16
Used cache with counter = 1
Used cache with counter = 2
Used cache with counter = 3
Used cache with counter = 4
Used cache with counter = 5
The function was executed 6 times
###
Still, the task is to count the number of function calls with the new argument and the number of cache results. We had the solution on one of the slides of the lecture presentation.

File: HW_decorator.py [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+print(exp_two(21))
+
+
+@repeat(6)
+def exp_two(num):
+    return num ** 2
+
+
+print(exp_two(4))
```
Comment (katevoskresenska) [inline HW_decorator.py:84]: It would be nice to add some documentation strings to all the functions.

**Review by mmykhailenko — APPROVED** [review]

**Pull request merged**